java-symbolic-execution/src/main/java/org/sonar/java/se/checks/XmlValidatedSighatureCheck.java

```diff
@@ -0,0 +1,221 @@
+    .names("setProperty")
+    .addParametersMatcher("java.lang.String", "java.lang.Object")
+    .build();
+
+  private static final MethodMatchers XML_SIGNATURE_VALIDATE = MethodMatchers.create()
+    .ofTypes("javax.xml.crypto.dsig.XMLSignature")
+    .names("validate")
+    .addParametersMatcher(JAVAX_XML_CRYPOT_VALIDATE_CONTEXT)
+    .build();
+
+  private enum DomSecureValidation implements Constraint {
+    DISABLED, EXPLICITLY_DISABLED;
+
+    private static final Predicate<Constraint> IS_EXPLICITLY_DISABLED = c -> c == EXPLICITLY_DISABLED;
+  }
+
+  private static final List<Class<? extends Constraint>> DOMAINS = Collections.singletonList(DomSecureValidation.class);
+
+  @Override
+  public ProgramState checkPostStatement(CheckerContext context, Tree syntaxNode) {
+    PostStatementVisitor visitor = new PostStatementVisitor(context);
+    syntaxNode.accept(visitor);
+    return visitor.programState;
+  }
+
+  private static class PostStatementVisitor extends CheckerTreeNodeVisitor {
+
+    private PostStatementVisitor(CheckerContext context) {
+      super(context.getState());
+    }
+
+    @Override
+    public void visitNewClass(NewClassTree tree) {
+      if (DOM_VALIDATE_CONTEXT_CONSTRUCTOR.matches(tree)) {
+        programState = programState.addConstraint(programState.peekValue(0), DomSecureValidation.DISABLED);
+      }
+    }
+
+    @Override
+    public void visitAssignmentExpression(AssignmentExpressionTree tree) {
+      ProgramState.SymbolicValueSymbol peek = programState.peekValueSymbol();
+      Symbol symbol = peek.symbol();
+      SymbolicValue sv = peek.symbolicValue();
+      if (symbol != null && sv instanceof DomValidateContextSymbolicValue) {
+        ((DomValidateContextSymbolicValue) sv).setField(ProgramState.isField(symbol));
+      }
+    }
+  }
+
+  @Override
+  public ProgramState checkPreStatement(CheckerContext context, Tree syntaxNode) {
+    PreStatementVisitor visitor = new PreStatementVisitor(context);
+    syntaxNode.accept(visitor);
+    return visitor.programState;
+  }
+
+  private class PreStatementVisitor extends CheckerTreeNodeVisitor {
+
+    private static final String SECURE_VALIDATION_PROPERTY = "org.jcp.xml.dsig.secureValidation";
+    private final CheckerContext context;
+
+    private PreStatementVisitor(CheckerContext context) {
+      super(context.getState());
+      this.context = context;
+    }
+
+    @Override
+    public void visitNewClass(NewClassTree newClass) {
+      if (DOM_VALIDATE_CONTEXT_CONSTRUCTOR.matches(newClass)) {
+        context.getConstraintManager().setValueFactory(() -> new DomValidateContextSymbolicValue(newClass.identifier()));
+      }
+    }
+
+    @Override
+    public void visitMethodInvocation(MethodInvocationTree mit) {
+      if (SET_PROPERTY.matches(mit)) {
+        Arguments args = mit.arguments();
+        if (!SECURE_VALIDATION_PROPERTY.equalsIgnoreCase(args.get(0).asConstant(String.class).orElse(""))) {
```

Comment:
What do you think about this code:
```suggestion
        if (args.get(0).asConstant(String.class).filter(SECURE_VALIDATION_PROPERTY::equalsIgnoreCase).isEmpty()) {
```
to avoid the negation and the possible comparison with an empty String?

Reply:
nice!